tools/mason/MasonArgParse.chpl

```diff
@@ -22,11 +22,29 @@ module MasonArgParse {
   private use Map;
   private use IO;
   private use Sort;
+  private use Sys only sys_getenv;
 
-  const DEBUG=false;
-  // TODO: Add positional arguments
-  // TODO: Add pass-thru options following "-" or "--"
+  config var OTFDebug=false;
+
+  const DEBUG = _checkDebug();
+
+  proc _checkDebug() : bool {
+    var envValue = "FALSE";
+    var rtn = false;
+
+   // if sys_getenv("ARGPARSE_DEBUG", envValue) == 1 ) {
+      if _convertStringToBool(envValue, rtn) {
+        return rtn || OTFDebug;
+      }
+   // }
+    return false;
+  }
```

Comment:
This debug code looks really sketchy. I'd consider just leaving it as a `config var` but if it were me I'd probably leave it as a `config param` or even just a `const DEBUG = false` and adjust it and recompile if I needed it.

Reply:
Could you elaborate on 'sketchy'? Is it error prone? Is there a bug in the logic? Not portable?
I put it there because I was hoping to read an environment variable, but failed to compile using the `sys_getenv` method. I left it commented out to remind me to ask about what I did wrong in trying to get it to work.

I do like having at least a config var for this because compilation takes so long, I felt like I was losing a lot of time recompiling just to set this value. I can remove the other code though, since it never worked anyway.

Reply:
Hi, I just meant by "sketchy" that it is "like a sketch" meaning it doesn't look all the way complete.

Removing the sys_getenv and relying on a config var works for me. You might consider making it a `private` config var so that it does not appear in the `--help` for programs using this. (IIRC private config vars can be set with the module name, e.g. `--MasonArgParse.DEBUG=true` but you might want to check me).

Reply:
Ah, excellent. I will clean that up and just use a `private` config var then.